joomla/joomla-cms
# [4.0] Database Driver Fixes

Pull Request for Issue #18660

### Summary of Changes

Because I seriously doubt the [Framework pull request](https://github.com/joomla-framework/database/pull/114) is going to get the attention it needs, I'm bringing the development branch the pull request is based on into the CMS as a PR in hopes that one or two people might actually test it instead of us having to resort to a blind merge and hope for the best scenario.

See the linked pull request for details about the actual code changes involved in this pull request, and the linked issue for the various issues exposed in the CMS environment that should be fixed with this pull request.

### Testing Instructions

When installing Joomla, depending on which PHP extensions are installed and enabled in your environment, you should be able to install Joomla on:

- MySQLi (requires PHP's `ext/mysqli` to be enabled, prior to this PR it required that the MySQL support be compiled with `mysqlnd` but now environments where the MySQL support is compiled with `libmysql` should be supported)
- PDO MySQL (requires PHP's `ext/pdo` with MySQL support to be enabled)
- PostgreSQL (requires PHP's `ext/pgsql` to be enabled)
- PDO PostgreSQL (requires PHP's `ext/pdo` with PostgreSQL support to be enabled)

If using either of the PDO drivers, you will most likely need https://github.com/joomla/joomla-cms/pull/19989 to be applied in your environment (if/when that PR is merged this PR will be rebased so the branch includes those changes).

Once installed, Joomla should be able to be used as normal (well, as normal as one can against the current 4.0 development state).
